Modules/_threadmodule.c

```diff
@@ -1148,20 +1092,17 @@ static PyObject *
 rlock_release_save(PyObject *op, PyObject *Py_UNUSED(ignored))
 {
     rlockobject *self = (rlockobject*)op;
-    PyThread_ident_t owner;
-    unsigned long count;
 
-    if (self->rlock_count == 0) {
+    if (!_PyRecursiveMutex_IsLockedByCurrentThread(&self->lock)) {
         PyErr_SetString(PyExc_RuntimeError,
                         "cannot release un-acquired lock");
         return NULL;
     }
 
-    owner = self->rlock_owner;
-    count = self->rlock_count;
-    self->rlock_count = 0;
-    _Py_atomic_store_ullong_relaxed(&self->rlock_owner, 0);
-    PyThread_release_lock(self->rlock_lock);
+    PyThread_ident_t owner = self->lock.thread;
+    size_t count = self->lock.level + 1;
+    self->lock.level = 0;  // ensure the unlock releases the lock
+    _PyRecursiveMutex_Unlock(&self->lock);
     return Py_BuildValue("k" Py_PARSE_THREAD_IDENT_T, count, owner);
```

Comment:
`"k"` is for `unsigned long`, but `count` is a `size_t`. I don't think we have a format specifier for `size_t`, but I think you can use `"n"` and cast `count` to a `Py_ssize_t`.

https://docs.python.org/3/c-api/arg.html#numbers

Reply:
thanks, fixed